tests/test_client.py

```diff
@@ -249,66 +211,295 @@ def test_get_crawl_results(client: Supadata, requests_mock) -> None:
+    assert channel.id == mock_response["id"]
+    assert channel.name == mock_response["name"]
+    assert channel.handle == mock_response["handle"]
+    assert channel.description == mock_response["description"]
+    assert channel.video_count == mock_response["videoCount"]
+    assert channel.subscriber_count == mock_response["subscriberCount"]
+    assert channel.thumbnail == mock_response["thumbnail"]
+    assert channel.banner == mock_response["banner"]
+
+
+def test_youtube_channel_invalid_id(client: Supadata, requests_mock) -> None:
+    channel_id = "UCsBjURrPoezyLs9EqgamOA"
+    mock_response = {
+        "error": "not-found",
+        "message": "The requested item could not be found",
+        "details": "The requested item could not be found.",
+    }
+    requests_mock.get(
+        f"{client.base_url}/youtube/channel?id={channel_id}",
+        status_code=404,
+        json=mock_response,
+    )
+    with pytest.raises(SupadataError) as exc_info:
+        client.youtube.channel(channel_id)
+
+    error = exc_info.value
+    assert error.error == "invalid-request"
+    assert error.message == "Endpoint does not exist"
+    assert error.details == "The requested item could not be found"
+
+
+def test_youtube_playlist(client: Supadata, requests_mock) -> None:
+    playlist_id = "PL0vfts4VzfNjQOM9VClyL5R0LeuTxlAR3"
+    mock_response = {
+        "id": playlist_id,
+        "title": "CS101",
+        "videoCount": 17,
+        "viewCount": 440901,
+        "lastUpdated": "2024-07-06T00:00:00.000Z",
+        "channel": {"id": "UCsBjURrPoezykLs9EqgamOA", "name": "Fireship"},
+    }
+
+    requests_mock.get(
+        f"{client.base_url}/youtube/playlist?id={playlist_id}", json=mock_response
+    )
+
+    playlist = client.youtube.playlist(playlist_id)
+    assert isinstance(playlist, YoutubePlaylist)
+    assert playlist.id == mock_response["id"]
+    assert playlist.title == mock_response["title"]
+    assert playlist.video_count == mock_response["videoCount"]
+    assert playlist.view_count == mock_response["viewCount"]
+    assert playlist.last_updated == datetime.fromisoformat(mock_response["lastUpdated"])
+    assert playlist.channel == mock_response["channel"]
+
+
+def test_youtube_playlist_invalid_id(client: Supadata, requests_mock) -> None:
+    playlist_id = "PL0vfts4VzfNjQOM9VClyL50LeuTxlAR3"
+    mock_response = {
+        "error": "not-found",
+        "message": "The requested item could not be found",
+        "details": "The requested item could not be found.",
+    }
+
+    requests_mock.get(
+        f"{client.base_url}/youtube/playlist?id={playlist_id}",
+        status_code=404,
+        json=mock_response,
+    )
+
+    with pytest.raises(SupadataError) as exc_info:
+        client.youtube.playlist(playlist_id)
+
+    error = exc_info.value
+    assert error.error == "invalid-request"
+    assert error.message == "Endpoint does not exist"
+    assert error.details == "The requested item could not be found"
```

Comment:
is the assertion correct?

the error and message seem to be different than the mock response